pulp_container/app/authorization.py

```diff
@@ -132,25 +132,24 @@ def determine_access(self):
         typ, name, actions = self.scope.split(":")
         actions = set(actions.split(","))
 
-        permitted_actions = set()
-        for action in actions:
-            has_permission = self.actions_permissions[action](name)
-            if has_permission:
-                permitted_actions.add(action)
+        permitted_actions = {action for action in actions if self.actions_permissions[action](name)}
 
         return {"type": typ, "name": name, "actions": list(permitted_actions)}
 
     def has_pull_permissions(self, path):
         """
         Check if the user has permissions to pull from the repository specified by the path.
         """
-        if self.user.has_perm("container.view_containerdistribution"):
+        if self.has_push_permissions(path):
```

Comment:
:+1:

Reply:
Actually this will double the checking for push_permission if you issue `["push","pull"]`. I tried to get the  tests pass that way.

Reply:
And, as expected, your last changes to the push pull permission logic fixed it for me.